frontend/src/router/pages/SubmitDates.jsx

```diff
@@ -160,32 +165,43 @@ function SubmitDates() {
           },
         ],
       },
-    });
+    }));
   }
 
-  function updateDateRange(dateableId, dateType, index, key, value) {
-    setDates({
-      ...dates,
-      [dateableId]: {
-        ...dates[dateableId],
-        [dateType]: dates[dateableId][dateType].map((dateRange, i) =>
-          i === index
-            ? { ...dateRange, [key]: formatDatetoISO(value), changed: true }
-            : dateRange,
-        ),
-      },
+  function updateDateRange(
+    dateableId,
+    dateType,
+    index,
+    key,
+    value,
+    callback = null,
+  ) {
+    const newValue = value ? value.toISOString() : null;
+
+    setDates((prevDates) => {
+      const updatedDates = cloneDeep(prevDates);
+
+      // Update the date value and mark the date range as changed
+      lodashSet(updatedDates, [dateableId, dateType, index, key], newValue);
+      lodashSet(updatedDates, [dateableId, dateType, index, "changed"], true);
```

Comment:
I was having so much trouble with stale state, I swapped this code out for Lodash - I don't think it made a difference but it's more readable anyway 🤷